src/App.tsx

```diff
@@ -14,11 +14,93 @@ const goodsFromServer: string[] = [
   'Garlic',
 ];
 
-const App: React.FC = () => (
-  <div className="App">
-    <h1>Selected good: -</h1>
-    {goodsFromServer.length}
-  </div>
-);
+interface State {
+  items: string[];
+}
+
+class App extends React.Component<{}, State> {
+  state: State = {
+    items: ['Jam'],
+  };
+
+  selectItem = (good: string) => {
+    this.setState((prevState) => ({ items: [...prevState.items, good] }));
+  };
+
+  removeItem = (good: string) => {
+    this.setState((prevState) => ({
+      items: prevState.items.filter(remaining => remaining !== good),
+    }));
+  };
+
+  buttonSelection = (isSelected: boolean, good: string) => {
+    return isSelected
+      ? this.removeItem(good)
+      : this.selectItem(good);
+  };
+
+  createMessage = () => {
+    const { items } = this.state;
+
+    const exceptLastGood = items.slice(0, -1).join(', ');
+    const lastGood = items.slice(-1);
+
+    switch (items.length) {
+      case 0:
+        return 'No goods selected';
+      case 1:
+        return `${items[0]} is selected`;
+      case 2:
+        return `${items[0]} and ${items[1]} are selected`;
+      default:
+        return (`${exceptLastGood} and ${lastGood} are selected`);
+    }
+  };
+
+  render(): React.ReactNode {
+    return (
+      <div className="App">
+        <div>
+          <h1>
+            {`Selected good: ${this.createMessage()}`}
+          </h1>
+
+          <ul>
+            {goodsFromServer.map((good: string) => {
+              const isSelected = this.state.items.includes(good);
+              const buttonText = isSelected ? 'Remove' : 'Select';
+
+              return (
+                <li key={good}>
+                  <span>
+                    {good}
+                  </span>
+
+                  <button
+                    type="button"
+                    onClick={() => this.buttonSelection(isSelected, good)}
+                  >
+                    {buttonText}
+                  </button>
+                </li>
+              );
+            })}
+          </ul>
+
+          {!!this.state.items.length && (
```

Comment:
I don't really understand what the `!!` does here.

Reply:
I think it's just a trick from OP to catch if I'm copying his code. He got me...

Reply:
I think it's just a trick from the OP to check if I'm copying his code. He got me...